sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -52,6 +53,15 @@ trait FunctionRegistry {
   /** Drop a function and return whether the function existed. */
   def dropFunction(name: String): Boolean
 
+  /* Return the FunctionBuilder and ExpressionInfo for the specified function name and classname. */
+  def makeFunctionBuilderAndInfo(
+    name: String,
+    functionClassName: String): (ExpressionInfo, FunctionBuilder) = {
+    val clazz = Utils.getContextOrSparkClassLoader.loadClass(functionClassName)
+    val (_, (info, builder)) =
+      FunctionRegistry.expression(name, clazz.asInstanceOf[Class[Expression]])
```

Comment:
We shouldn't expose this to the user. The reflection hacks we do down there are intended to be for internal use only so we don't want the user to do something like:
```
CREATE TEMPORARY FUNCTION my_func AS com.x.y.z.MyFunc
```
where `MyFunc` has the special `Expression` constructors that we expect.